tests/client/test_users.py

```diff
@@ -0,0 +1,92 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from uuid import UUID
+
+import pytest
+from argilla.client.api import ArgillaSingleton
+from argilla.client.users import User
+from argilla.server.models import User as UserModel
+
+from tests.factories import UserFactory
+
+
+def test_user_cls_init() -> None:
+    with pytest.raises(
+        Exception,
+        match=r"`User` cannot be initialized via the `__init__` method | you should use `User.from_name\('test_user'\)`",
+    ):
+        User(name="test_user")
+
+    with pytest.raises(
+        Exception,
+        match=r"`User` cannot be initialized via the `__init__` method | you should use `User.from_id\('00000000-0000-0000-0000-000000000000'\)`",
+    ):
+        User(id="00000000-0000-0000-0000-000000000000")
+
+
+def test_user_from_name(owner: UserModel):
```

Comment:
Maybe is more clear as ServerUser
```suggestion
def test_user_from_name(owner: ServerUser):
```